example/tests/test_serializers.py

```diff
@@ -193,3 +198,40 @@ def test_model_serializer_with_implicit_fields(self, comment, client):
 
         assert response.status_code == 200
         assert expected == response.json()
+
+
+class TestPolymorphicModelSerializer(TestCase):
+    def setUp(self):
+        self.project = ArtProjectFactory.create()
+
+    def test_polymorphic_model_serializer_passes_instance_to_child(self):
+        """
+        Ensure that `PolymorphicModelSerializer` is passing the instance to the
+        child serializer when initializing them
+        """
+        # Arrange
+        self.serializer_instance = None
+
+        def save(serializer_self):
+            """
+            Override `ArtProjectSerializer.save` to get the instance serializer,
+            which is later used assertion
+            """
+            self.serializer_instance = serializer_self.instance
```

Comment:
I would add the asserts right here. This way no additional variables are needed. Additionally can you also assert for serializer_self.data and serializer_self.partial as well for completeness.